lib/PrintAsClang/CxxSynthesis.h

```diff
@@ -0,0 +1,40 @@
+//===--- CxxSynthesis.h - Rules for synthesizing C++ code -------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_PRINTASCLANG_CXXSYNTHESIS_H
+#define SWIFT_PRINTASCLANG_CXXSYNTHESIS_H
+
+#include "swift/Basic/LLVM.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/raw_ostream.h"
+
+namespace swift {
+
+namespace cxx_synthesis {
```

Comment:
I don't think there's anything wrong with this, but usually, there'd be some object which we put all these functions in. Maybe `Implementation`?

Reply:
Sure, I ca pull it into an object class.